pkgs/development/tools/misc/unixbench/default.nix

```diff
@@ -0,0 +1,103 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, makeWrapper
+, pandoc
+, installShellFiles
+, perl
+, xorg
+, libGL
+, bash
+, coreutils
+, nettools
+, glibc
+, gcc
+, gnugrep
+, gawk
+, withGL? true
+, withX11perf? true
+}:
+
+stdenv.mkDerivation rec {
+  pname = "unixbench";
+  version = "unstable-2023-02-27";
+
+  src = fetchFromGitHub {
+    owner = "kdlucas";
+    repo = "byte-unixbench";
+    rev = "81e9de58c50566da8ae45c0ea7a55e7a7066ea75";
+    hash = "sha256-mAgcwvfYDrPYqrjTzlRBYlLo7DK93CPi2iavB5+oagk=";
+  };
+
+  patches = [
+    ./common.patch
+  ];
+
+  patchFlags = [ "-p2" ];
+
+  sourceRoot = "source/UnixBench";
+
+  nativeBuildInputs = [
+    makeWrapper
+    pandoc
+    installShellFiles
+  ];
+
+  buildInputs = [ perl ] ++ lib.optionals withGL [
+    xorg.libX11
+    xorg.libXext
+    libGL
+  ];
+
+  runtimeDependencies = [
+    coreutils
+    nettools
+    glibc
+    gcc
```

Comment:
🤯

Reply:
It calls these binaries, so for reproducing purposes...

Reply:
I guess these should be prefixed with `buildPackages`, if we really want to...uh...do that. And for neutrality, `gcc` here should be `buildPackages.stdenv.cc`, and `glibc` should be `buildPackages.stdenv.cc.libc`.

Reply:
> Reference in new

I'm not quite sure about it. Does that mean the platform of these packages matches the target platform of unixbench?

Reply:
Also the script uses glibc binaries, `locale`. It needs gcc to compile a test c file in runtime, for benchmarking purposes (In this case, specifying one compiler all the time should make more sense). I can change `gcc` to `cc` If it's proper to do that.

https://github.com/kdlucas/byte-unixbench/blob/a07fcc03264915c624f0e4818993c5b4df3fa703/UnixBench/Run#L84

Reply:
> > Reference in new
> 
> I'm not quite sure about it. Does that mean the platform of these packages matches the target platform of unixbench?

No, not `buildPackages`, since the case to avoid is having a `cc` running on `buildPlatform`. I think it should be `pkgsHostHost`, but I could still be wrong.

Reply:
> Also the script uses glibc executable `locale`. It needs gcc to compile a test c file in runtime, for benchmarking purposes (In this case, specifying one compiler all the time should make more sense). I can change `gcc` to `cc` If it's proper to do that.

`locale` on Darwin is elsewhere; in fact, `glibc` does not even support Darwin. Just use `locale` instead.

https://github.com/NixOS/nixpkgs/blob/9daee206706752c84a40ad207f88d4231a2c9ebd/pkgs/top-level/unixtools.nix#L107-L110

Reply:
> (But this is more bizarre: why not use `pkgs.darwin.adv_cmds` on Darwin instead?)

I'm a poor student I don't own any Apple device I don't understand nix-darwin please forgive me 🥺

Reply:
> No, not `buildPackages`, since the case to avoid is having a `cc` running on `buildPlatform`. I think it should be `pkgsHostHost`, but I could still be wrong.

It should be `pkgsHostTarget` since the `gcc` is supposed to compile that simple file for the same architecture, which means the host and target platform of it matches the host platform of unixbench (similar to [buildInputs](https://ryantm.github.io/nixpkgs/stdenv/stdenv/#var-stdenv-buildInputs)), and we only need to consider that far for gcc because it has ability to do cross compilation, am I right?

I have a headache now.

Reply:
> which means the host and target platform of it matches the host platform of unixbench

This is exactly what `pkgsHostHost` guarantees.